packages/internal-playwright-helpers/src/pages/testApp.ts

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import { Page } from "@playwright/test";
+import {
+  TESTID_OPENID_PROVIDER_INPUT,
+  TESTID_LOGIN_BUTTON,
+  TESTID_ERROR_MESSAGE,
+  TESTID_SESSION_STATUS
+} from "@inrupt/internal-playwright-testids";
+
+export class TestPage {
+  page: Page;
+  openidProvider: string;
+
+  constructor(page: Page, openidProvider: string) {
+    this.page = page;
+    this.openidProvider = openidProvider;
+  }
+
+  async start() {
+    await this.page.goto("/");
+  }
+
+  async startLogin() {
+    await this.page.fill(
+      `[data-testid=${TESTID_OPENID_PROVIDER_INPUT}]`,
+      this.openidProvider
+    );
+    await Promise.all([
+      // It is important to call waitForNavigation before click to set up waiting.
+      this.page.waitForNavigation(),
+      // Clicking the link will indirectly cause a navigation.
+      this.page.click(`[data-testid=${TESTID_LOGIN_BUTTON}]`),
```

Comment:
would it be worth the testids package exporting these selectors itself? e.g.,

```suggestion
      this.page.click(selectors.TESTID_LOGIN_BUTTON),
```

Instead

Reply:
The package will need to export the raw testids in any case since they need to be imported by the test app, so I'm not sure also exporting the selectors would add much. So you think it would improve readability?

Reply:
Yeah, it's really abouf readability here & to ensure the testid isn't accidentally a prefix instead of the complete value

Reply:
5b8f4f8